pkg/actions/addon/addon.go

```diff
@@ -43,6 +44,12 @@ func New(clusterConfig *api.ClusterConfig, eksAPI awsapi.EKS, stackManager manag
 }
 
 func (a *Manager) waitForAddonToBeActive(ctx context.Context, addon *api.Addon, waitTimeout time.Duration) error {
+	pool := cmdutils.ToNodePools(a.clusterConfig)
+	// We don't wait for coredns if there are no nodegroups. It will get into degraded state
+	// and recover once nodegroups are added.
+	if len(pool) == 0 && addon.Name == api.CoreDNSAddon {
+		return nil
+	}
```

Comment:
```suggestion
	// We don't wait for coredns if there are no nodegroups. It will get into degraded state
	// and recover once nodegroups are added.
	if addon.Name == api.CoreDNSAddon && (len(a.clusterConfig.NodeGroups) > 0 || len(a.clusterConfig.ManagedNodeGroups) > 0) {
      return nil
	}
```

Reply:
Fine. :D :D :D